runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/RequiredParametersPresent.java

```diff
@@ -0,0 +1,115 @@
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import static java.lang.String.format;
+import static java.util.Optional.of;
+import static java.util.stream.Stream.concat;
+import static org.mule.runtime.ast.api.util.ComponentAstPredicatesFactory.currentElemement;
+import static org.mule.runtime.ast.api.validation.Validation.Level.ERROR;
+import static org.mule.runtime.ast.api.validation.ValidationResultItem.create;
+
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.api.meta.model.source.SourceCallbackModel;
+import org.mule.runtime.api.meta.model.source.SourceModel;
+import org.mule.runtime.api.util.Pair;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.ComponentParameterAst;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+@IgnoreOnLazyInit(forceDslDeclarationValidation = true)
+public class RequiredParametersPresent implements Validation {
+
+  @Override
+  public String getName() {
+    return "Required parameters are present.";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Required parameters are present.";
+  }
+
+  @Override
+  public Level getLevel() {
+    return ERROR;
+  }
+
+  @Override
+  public Predicate<List<ComponentAst>> applicable() {
+    return currentElemement(comp -> comp.getModel(ParameterizedModel.class)
+        .map(pmzd -> pmzd.getParameterGroupModels().stream()
+            .flatMap(pmg -> pmg.getParameterModels().stream())
+            .anyMatch(this::isDoValidation))
+        .orElse(false));
+  }
+
+  @Override
+  public Optional<ValidationResultItem> validate(ComponentAst component, ArtifactAst artifact) {
+    return requiredParamsStream(component)
+        .map(param -> {
+          if (param.getSecond() == null || !param.getSecond().getValue().getValue().isPresent()) {
+            return of(create(component, this,
+                             format("Element <%s> is missing required parameter '%s'.",
+                                    component.getIdentifier().toString(),
+                                    param.getFirst().getName())));
+          } else if (param.getSecond().getValue().getRight() instanceof ComponentAst) {
+            // validate any nested pojos as well...
+            return validate((ComponentAst) param.getSecond().getValue().getRight(), artifact);
+          } else {
+            return Optional.<ValidationResultItem>empty();
+          }
+        })
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .findFirst();
+  }
+
+  // resolve the provided value for every required param to iterate and check
```

Comment:
use appropriate comment format for a method or add this this comment in the method.